src/blazingmq/_session.py

```diff
@@ -55,19 +55,33 @@ def DefaultMonitor() -> Union[BasicHealthMonitor, None]:
 KNOWN_MONITORS = ("blazingmq.BasicHealthMonitor",)
 
 
-def _convert_timeout(timeout: float) -> Optional[float]:
+def _convert_timeout(timeout: Optional[float]) -> Optional[float]:
     """Convert the timeout for use by the Cython layer.
 
-    If it is the DEFAULT_TIMEOUT sentinel, return None.  Otherwise, validate
-    that it is within the range accepted by bsls::TimeInterval and return it.
+    If it is the DEFAULT_TIMEOUT sentinel or None, return None.  Otherwise,
+    validate that it is within the range accepted by bsls::TimeInterval and
+    return it.
     """
-    if timeout is DEFAULT_TIMEOUT:
+    if timeout is DEFAULT_TIMEOUT or timeout is None:
         return None
-    elif timeout is not None and (0.0 < float(timeout) < 2**63):
-        return float(timeout)
+    elif 0.0 < timeout < 2**63:
```

Comment:
`sys.maxsize` might be preferred to `2**63`

Reply:
On second thought, I'm not sure.  This function checks that the number of seconds we pass in can be represented by the `bsls::TimeInterval` our C++ code uses.  [`bsls::TimeInterval` can store up to `2**63-1` seconds](https://bloomberg.github.io/bde-resources/doxygen/bde_api_prod/classbsls_1_1TimeInterval.html), regardless of the system pointer size.  So I think we do really want this constant.

Reply:
Okay sure. Using `2**63` is more explicit.